internal/server/iigointernal/judiciary.go

```diff
@@ -0,0 +1,196 @@
+	executiveBranch.budget = amount
+}
+
+// PayPresident pays the president salary.
+func (j *judiciary) PayPresident() (shared.Resources, error) {
+	hold := j.presidentSalary
+	j.presidentSalary = 0
+	return hold, nil
+}
+
+// setSpeakerAndPresidentIDs set the speaker and president IDs.
+func (j *judiciary) setSpeakerAndPresidentIDs(speakerId shared.ClientID, presidentId shared.ClientID) {
+	j.speakerID = speakerId
+	j.presidentID = presidentId
+}
+
+// InspectHistory checks all actions that happened in the last turn and audits them.
+// This can be overridden by clients.
+func (j *judiciary) inspectHistory() (map[shared.ClientID]roles.EvaluationReturn, error) {
+	j.budget -= serviceCharge
+	return j.clientJudge.InspectHistory()
+}
+
+// inspectBallot checks each ballot action adheres to the rules
+func (j *judiciary) inspectBallot() (bool, error) {
+	// 1. Evaluate difference between newRules and oldRules to check
+	//    rule changes are in line with RuleToVote in previous ballot
+	// 2. Compare each ballot action adheres to rules in ruleSet matrix
+	j.budget -= serviceCharge // will be removed post-MVP
+	rulesAffectedBySpeaker := j.EvaluationResults[j.speakerID]
+	indexOfBallotRule, err := searchForRule("inspect_ballot_rule", rulesAffectedBySpeaker.Rules)
+	if err == nil {
+		return rulesAffectedBySpeaker.Evaluations[indexOfBallotRule], nil
+	} else {
+		return true, errors.Errorf("Speaker did not conduct any ballots")
+	}
+}
+
+// inspectAllocation checks each resource allocation action adheres to the rules
+func (j *judiciary) inspectAllocation() (bool, error) {
+	// 1. Evaluate difference between commonPoolNew and commonPoolOld
+	//    to check resource allocation changes are in line with ResourceRequests
+	//    in previous resourceAllocation
+	// 2. Compare each resource allocation action adheres to rules in ruleSet
+	//    matrix
+	j.budget -= serviceCharge // will be removed post-MVP
+	rulesAffectedByPresident := j.EvaluationResults[j.presidentID]
+	indexOfAllocRule, err := searchForRule("inspect_allocation_rule", rulesAffectedByPresident.Rules)
+	if err == nil {
+		return rulesAffectedByPresident.Evaluations[indexOfAllocRule], nil
+	} else {
+		return true, errors.Errorf("President didn't conduct any allocations")
+	}
+}
+
+// searchForRule searches for a given rule in the RuleMatrix
+func searchForRule(ruleName string, listOfRuleMatrices []rules.RuleMatrix) (int, error) {
+	for i, v := range listOfRuleMatrices {
+		if v.RuleName == ruleName {
+			return i, nil
+		}
+	}
+	return 0, errors.Errorf("The rule name '%v' was not found", ruleName)
+}
+
+// declareSpeakerPerformanceWrapped wraps the result of DeclareSpeakerPerformance for orchestration
+func (j *judiciary) declareSpeakerPerformanceWrapped() {
+
+	result, checkRole := j.clientJudge.DeclareSpeakerPerformance()
+
+	message := generateSpeakerPerformanceMessage(j.BallotID, result, j.speakerID, checkRole)
+	broadcastToAllIslands(shared.TeamIDs[j.JudgeID], message)
+
+}
```

Comment:
Nitpick: Theres a lot of new lines which aren't needed

Reply:
sorted!